onnxruntime/core/providers/cuda/tensor/space_depth_ops.cc

```diff
@@ -142,42 +225,65 @@ Status DepthToSpace::ComputeInternal(OpKernelContext* context) const {
                                                         batch,
                                                         input_depth, input_height, input_width,
                                                         output_depth, output_height, output_width,
-                                                        false));
+                                                        false, NHWC));
 
   // We use the "actual" output shape to construct the output tensor
-  Tensor& output = *context->Output(0, {batch, output_depth, output_height, output_width});
-
+  Tensor* output;
   // We will pass in the "virtual" output shape to be used by DoTranspose() in SpaceDepthOpCudaImpl(...)
-  TensorShape virtual_output_shape{batch, input_depth / blocksize_ / blocksize_,
+  TensorShape virtual_output_shape;
+  if (NHWC) {
+    output = context->Output(0, {batch, output_height, output_width, output_depth});
+    virtual_output_shape = {batch, input_height, blocksize_, input_width, blocksize_,
+                            input_depth / blocksize_ / blocksize_};
+  } else {
+    output = context->Output(0, {batch, output_depth, output_height, output_width});
+    virtual_output_shape = {batch, input_depth / blocksize_ / blocksize_,
                                    input_height, blocksize_, input_width, blocksize_};
+  }
 
   std::vector<size_t> permutation;
   permutation.reserve(6);
   permutation.push_back(0);
```

Comment:
you could have one static constexpr for each permutation to avoid construction it on each run. I hope cuDNN accepts const std::vector